pkg/kube/inject/webhook.go

```diff
@@ -432,6 +432,32 @@ func escapeJSONPointerValue(in string) string {
 	return strings.Replace(step, "/", "~1", -1)
 }
 
+func updateLabels(target map[string]string, added map[string]string) (patch []rfc6902PatchOperation) {
+	for key, value := range added {
+		if target == nil {
+			target = map[string]string{}
+			patch = append(patch, rfc6902PatchOperation{
+				Op:   "add",
+				Path: "/metadata/labels",
+				Value: map[string]string{
+					key: value,
+				},
+			})
+		} else {
+			op := "add"
+			if target[key] != "" {
+				op = "replace"
```

Comment:
should have a unit test case for this?

Reply:
Plus we shouldn't override if the label exists and equals to false set by users.

Other invalid (foo/bar) values might be able to be overriden to true.

And test cases for those.

Reply:
see comment below - I followed the `updateAnnotation` method to try and keep the function universal so it could be usable for other labels down the line

Reply:
if the caller is responsible to ensure the same `label` does not exists before, this `if target[key] != ""` can never be true. Correct?

I see the `updateAnnotations` above, but its caller does not have the check as the labels you're doing.

 think we should remove this line and not having check in two places. Rename the func to be `addLabel` to make it clearer.